portiaui/app/models/base.js

```diff
@@ -3,8 +3,110 @@ import DS from 'ember-data';
 import EmptyObject from "ember-data/-private/system/empty-object";
 const { get } = Ember;
 
+let currentActionModel = null;
+const modelActionQueue = [];
+
+function runActions() {
+    if (currentActionModel && currentActionModel.get('isSaving')) {
+        // current action has not finished
+        return;
+    }
+    currentActionModel = null;
+    while (modelActionQueue.length) {
+        const nextModel = modelActionQueue.shift();
+        if (nextModel.get('isSaving')) {
+            // model has started saving again, wait for the next didCommit event
+            return;
+        }
+        if (nextModel.get('isDeleted') && !nextModel.get('hasDirtyAttributes')) {
+            // model has been deleted
+            continue;
+        }
+        if (nextModel.get('pendingDelete')) {
+            nextModel.deleteRecord();
+            nextModel.set('pendingDelete', false);
```

Comment:
deleteRecord marks it as deleted, destoryRecord calls deleteRecord followed by save.
we don't continue here so that we can proceed to call save if there is a pending save too.
Maybe this needs a comment.